forms-flow-api/src/formsflow_api/resources/user.py

```diff
@@ -0,0 +1,102 @@
+"""Resource to call Keycloak Service API calls and filter responses."""
+from http import HTTPStatus
+
+from flask import current_app, g, request
+from flask_restx import Namespace, Resource
+from marshmallow import ValidationError
+
+from formsflow_api.schemas import UserlocaleReqSchema
+from formsflow_api.services import KeycloakAdminAPIService
+from formsflow_api.utils import auth, cors_preflight, profiletime
+
+API = Namespace("user", description="Keycloak user APIs")
+
+
+@cors_preflight("PUT, OPTIONS")
+@API.route("/locale", methods=["OPTIONS", "PUT"])
+class KeycloakUserService(Resource):
+    """Provides api interface for interacting with Keycloak user attributes."""
+
+    def __init__(self, *args, **kwargs):
+        """Initializing client."""
+        super().__init__(*args, **kwargs)
+        self.client = KeycloakAdminAPIService()
+
+    @auth.require
+    @profiletime
+    def __get_user_data(self) -> dict:
+        """GET the keycloak users based on the username and email params."""
+        user_name = g.token_info.get("preferred_username")
+        email = g.token_info.get("email")
+        if email is not None:
+            url_path = f"users?username={user_name}&email={email}&exact={True}"
+        else:
+            url_path = f"users?username={user_name}"
+        response = self.client.get_request(url_path=url_path)
+        response = response[0]
+        if response is None:
+            return {"message": "User not found"}, HTTPStatus.NOT_FOUND
+        return response
+
+    @auth.require
+    @profiletime
+    def put(self) -> dict:
```

Comment:
We may want to review this endpoint to make sure it does not have security implications. It'll be interesting to see what fields a user can update about his/her profile. Whether email/IDIR or user group/role are one of them?

We can also disable it until there is a need for it.